syntax ="proto3";
package auth;

service AuthService {
  rpc Login (LoginRequest) returns (LoginResponse);
  rpc Signup (SignupRequest) returns (SignupResponse);
  rpc CreateBooking (BookingRequest) returns (BookingResponse);
}

message LoginRequest {
  string username =1;
  string password =2;
}

message LoginResponse {
  bool success=1;
  string message=2;
  string token =3;
}

message SignupRequest {
  string username =1;
  string password =2;
}

message SignupResponse {
  bool success=1;
  string message =2;
}

// Booking messages
message BookingRequest {
  string username = 1;
  string train_number = 2;
  int32 qty = 3;
}

message BookingResponse {
  bool success = 1;
  string message = 2;
}